### DISCUSSION TITLE
F5-TTS-ONNX standalone install with simple gui

### DISCUSSION THREAD

**Discussion by patientx** [SWivid/F5-TTS#564] [Show and tell]
With the help of DakeQQ's [F5-TTS-ONNX](https://github.com/DakeQQ/F5-TTS-ONNX)  ,  (https://github.com/SWivid/F5-TTS/issues/2149) , I managed to install and use F5-TTS with my AMD gpu on windows with directml. (it is at least 5 to 8 times faster than cpu for me) Since I wanted a bit of ease-of-use I coded a simple gui and while doing it made it standalone with python venv usage, it auto installs everything, downloads ONNX models from huggingface and doesn't need the original models. It saves the values everytime you change it etc... Please give it a try, I am not a programmer and this was firstly for my personal use as a hobby so treat it as such :)

https://github.com/patientx/F5-TTS-ONNX-gui

**Comment by MithrilMan** [comment]
Interesting, as I've successfully implemented my finetune, my next step will be to convert to onnx because I want to use it with OnnxRuntime in C#

Do you have an onnx model available (doesn't matter which language) so I can test if I can make it run?

**Reply by patientx** [reply]
Model links are in the github. Both default model and the one I converted to allow longer audio. (degrades quality if too long)

**Reply by MithrilMan** [reply]
thanks, what do you mean by "allow longer audio", the reference audio or the total length?
what's the technique to increase limits?
I'm trying to finetune a model and I've quite an hard time to understand all the parameters involved and I'm not happy about wasting time renting GPUs without getting the most out of them (I just tried to rent an H100 but the few tests I did to setup gradio finetuning parameters are like shoting in the dark)

**Reply by patientx** [reply]
MAX_SIGNAL_LENGTH = 2048

**Reply by patientx** [reply]
> MAX_SIGNAL_LENGTH = 2048. When I first tried the onnx model with a sample audio this error kept popping up, it turned out my reference audio was a bit longer so I had to cut it shorter. But there is also another way, to increase this value while converting model to onnx. Now longer source and generated audio is possible. (But not perfect). After 20 seconds it tends to turn the last words into slurs :)

**Comment by heirofsihgma** [comment]
Very good! Can you also use this for Fine-Tuning or Training Voice Models or Checkpoints or LoRAs?

**Reply by patientx** [reply]
I don't think so I am afraid. Haven't looked into it in a while too.
